src/main/java/jenkins/branch/OrganizationFolder.java

```diff
@@ -337,5 +619,260 @@ public String getDisplayName() {
             return Messages.OrganizationFolder_OrganizationScan_displayName(getParent().getPronoun());
         }
 
+        @Override
+        public void run() {
+            long start = System.currentTimeMillis();
+            try {
+                super.run();
+            } finally {
+                long end = System.currentTimeMillis();
+                LOGGER.log(Level.INFO, "{0} #{1,time,yyyyMMdd.HHmmss} organization scan action completed: {2} in {3}",
+                        new Object[]{
+                                getParent().getFullName(), start, getResult(), Util.getTimeSpanString(end - start)
+                        }
+                );
+            }
+        }
+
+    }
+
+    @Extension
+    public static class SCMEventListenerImpl extends SCMEventListener {
+
+        public TaskListener globalEventsListener() {
+            File eventsFile =
+                    new File(Jenkins.getActiveInstance().getRootDir(), OrganizationFolder.class.getName() + ".log");
+            boolean rotate = eventsFile.length() > 30 * 1024;
+            OutputStream os = new ReopenableRotatingFileOutputStream(eventsFile, 5);
```

Comment:
We need to introduce some sort of reusable utility for this, not least because it will all need to be rewritten for @oleg-nenashev’s `LoggingMethod` changes.